e2e/cypress/integration/enterprise/guest_accounts/guest_experience_ui_spec.js

```diff
@@ -15,6 +15,17 @@
  */
 import * as TIMEOUTS from '../../../fixtures/timeouts';
 
+function demoteGuestUser(guestUser) {
+    // # Demote user as guest user before each test
+    cy.apiAdminLogin();
+    cy.apiGetUserByEmail(guestUser.email).then((emailResponse) => {
```

Comment:
`cy.apiAdminLogin()` and `cy.apiGetUserByEmail` are not necessary since `guestUser` is a user profile object with `id` on it.

Reply:
@saturninoabril I am trying to fetch guestUser object again as this might have been changed within the test, especially the second test. Hence if the guestUser was promoted as Regular member in a previous test, I am demoting the user as guest again. This is intentional.

Reply:
Honestly, the conditional only complicate things IMO. I'm on the fence of not caring whatever roles it may have in the first place if all I need is to ensure that the user is guest before the test run on specific case.
```javascript
if (user.roles !== 'system_guest') {
    cy.demoteUser(guestUser.id);
}
```

Reply:
This is handled because if we try to promote a user who is already a guest, the external request throws an error, which we do not want. Hence this specific check. 😄

Reply:
What error are you seeing? I tried it locally and works fine for me. Probably needs to login as admin before creating a team.

Reply:
@saturninoabril This is the error the API returns if the user is already a guest user and if we try to demote the user as guest again. 

```ERRO[2020-07-03T18:38:17.126246+05:30] Unable to convert the user to guest because is already a guest.  caller="mlog/log.go:175" err_details= err_where=Api4.demoteUserToGuest http_code=501 ip_addr=127.0.0.1 method=POST path=/api/v4/users/an5hee4of7r9tdicee6yrscfsa/demote request_id=jub94jmdrtbgif9qnaawcgepzc user_id=d4ymchskjf8xjp4k5hudnn37uc```

![Screenshot 2020-07-03 at 6 39 02 PM](https://user-images.githubusercontent.com/1429138/86472243-7e30f500-bd5c-11ea-9e82-74cf28392820.png)

Reply:
Oh you're right. Thanks for checking. Now I got the same error as you posted. I'm good as is.